lintro/ai/providers/cursor.py

```
@@ -0,0 +1,253 @@
+        try:
+            data = json.loads(stdout.strip())
+        except json.JSONDecodeError as e:
+            raise AIProviderError(
+                f"Cursor CLI returned invalid JSON: {e}\n"
+                f"Raw output: {stdout[:500]}",
+            ) from e
+
+        if data.get("is_error") or data.get("subtype") == "error":
+            raise AIProviderError(
+                f"Cursor CLI reported error: {data.get('result', stdout[:500])}",
+            )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)